modules/common/topology/topology.go

```diff
@@ -52,12 +55,33 @@ func EnsureTopologyRef(
 	if err != nil {
 		return topology, hash, err
 	}
+
 	// Add finalizer (if not present) to the resource consumed by the Service
 	if controllerutil.AddFinalizer(topology, fmt.Sprintf("%s-%s", h.GetFinalizer(), finalizer)) {
 		if err := h.GetClient().Update(ctx, topology); err != nil {
 			return topology, hash, err
 		}
 	}
+
+	// Set default LabelSelector on topologyConstraints if not set, similar to cluster level default:
+	// https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/#cluster-level-default-constraints
+	topology = topology.DeepCopy()
+
+	topologyConstraints := topology.Spec.TopologySpreadConstraints
+	if topologyConstraints != nil {
```

Comment:
could be `if topologyConstraints != nil && defaultLabelSelector != nil` ?

Reply:
sure, not at all worried about performance here though. Typically will have one or two topologyConstraints. Much bigger performance concern I expect is how slow k8s scheduler gets if there are many.